services/blockchain-connector/shared/sdk/client.js

```diff
@@ -56,22 +58,25 @@ const checkIsClientAlive = async () =>
 
 		if (
 			config.isUseLiskIPCClient ||
-			Date.now() - lastClientAliveTime < CLIENT_ALIVE_ASSUMPTION_TIME
+			Date.now() - lastClientAliveTime < CLIENT_ALIVE_ASSUMPTION_TIME ||
+			Date.now() - heartbeatCheckBeginTime < HEARTBEAT_ACK_MAX_WAIT_TIME
 		) {
 			return resolve(clientCache._channel && clientCache._channel.isAlive);
 		}
 
+		heartbeatCheckBeginTime = Date.now();
 		const boundPongListener = () => pongListener(resolve);
 
-		clientCache._channel._ws.on('pong', boundPongListener);
-		clientCache._channel._ws.ping(() => {});
+		const wsInstance = clientCache._channel._ws;
+		wsInstance.on('pong', boundPongListener);
+		wsInstance.ping(() => {});
```

Comment:
`isClientAlive` should be reset to `false` before each ping. Otherwise, it would be always `true` after first pong.

```suggestion
		isClientAlive = false;
		wsInstance.ping(() => {});
```